lib/RGBot.js

```diff
@@ -583,196 +1098,360 @@ const RGBot = class {
+                                    this.#bot.pathfinder.setGoal(null);
+                                    this.#bot.pathfinder.stop();
+                                } catch (err) {
+                                    this.#logDebug(`Caught expected goal changed error while stopping dig path`);
+                                }
+                            }
                         }
                     }
+                    this.#bot.on('path_reset', checkForInfiniteDig);
+                    this.#logDebug(`Digging ${this.getEntityName(block)}`);
+                    await this.#bot.dig(block);
+                    this.#bot.off('path_reset', checkForInfiniteDig);
+                    return true
+                } catch (ex) {
+                    this.#logWarn(`Error trying to equip tool or dig block ${this.getEntityName(block)}`, ex)
                 }
+            } else {
+                this.#log(`Cannot dig block ${this.getEntityName(block)}, no capable tool`)
             }
-            this.#bot.on('path_reset', checkForInfiniteDig);
-            this.#log(`Digging ${this.getEntityName(block)}`);
-            await this.#bot.dig(block);
-            this.#bot.off('path_reset', checkForInfiniteDig);
-            result = true;
         }
-        return result;
+        return false
     }
 
     /**
      * Locate and dig the closest Block of a given type within a maximum distance from the Bot.
-     * This method will equip the most appropriate tool in the Bot's inventory for this Block type.
+     * This method will equip the most appropriate tool in the bot inventory for this Block type.
      * @param {string} blockType The name of the Block to find and dig
-     * @param {object} options={} Optional parameters
+     * @param {object} [options={}] Optional parameters
      * @param {boolean} [options.partialMatch=false] Find blocks whose name / displayName contains blockType. (Ex. 'log' may find any of 'spruce_log', 'oak_log', etc.)
      * @param {boolean} [options.onlyFindTopBlocks=false] Will not attempt to dig any Blocks that are beneath another Block
      * @param {number} [options.maxDistance=50] Find any Blocks matching the search criteria up to and including this distance from the Bot
      * @param {boolean} [options.skipCollection=false] If true, the Bot will not explicitly attempt to collect drops from the broken Block. This allows the player to control which drops are collected and which ones are ignored
      * @param {boolean} [options.skipClosest=false] Will attempt to locate the next-closest Block. This can be used to skip the closest Block when the Bot encounters an issue collecting it
      * @returns {Promise<boolean>} true if a Block was found and dug successfully or false if a Block was not found or if digging was interrupted
+     *
+     * @deprecated Since 1.2.0, Collection should be a separate bot action/choice from digging. please use the example for new code
```

Comment:
I notice we already had a skipCollection option in the original API as well - any reason we couldn't use that for the new functionality?